pkg/awsutils/awsutils.go

```diff
@@ -740,11 +740,14 @@ func (cache *EC2InstanceMetadataCache) awsGetFreeDeviceNumber() (int, error) {
 
 // AllocENI creates an ENI and attaches it to the instance
 // returns: newly created ENI ID
-func (cache *EC2InstanceMetadataCache) AllocENI(useCustomCfg bool, sg []*string, subnet string) (string, error) {
-	eniID, err := cache.createENI(useCustomCfg, sg, subnet)
+func (cache *EC2InstanceMetadataCache) AllocENI(useCustomCfg bool, sg []*string, subnet string, numIPs int) (string, error) {
+	eniID, err := cache.createENI(useCustomCfg, sg, subnet, numIPs)
 	if err != nil {
 		return "", errors.Wrap(err, "AllocENI: failed to create ENI")
 	}
+	if eniID == "" {
+		return "", nil
+	}
```

Comment:
This is for the case that `needIPs < 1`. createENI would return `"", nil` since no ENI is created. I'm not sure if combining with above case makes sense since it technically isn't an error.

Reply:
I see, let's at least log and/or comment then